### PR TITLE
refactor(DB): Adjust table/column names to consistent standards.

### LINKED ISSUE #11005: Feature: Align DB column names to TrinityCore
### Describe your feature request or suggestion in detail

In line with https://github.com/azerothcore/azerothcore-wotlk/issues/1118, there's a few column discrepancies between azerothcore and trinitycore that come down to just different naming conventions. 

I've been asked to post this issue as a step to bringing AzerothCore closer to the TSWoW project. For our sake, these specific changes are **not really necessary anymore** as we currently [apply a translation layer between ac and tc](https://github.com/tswow/tswow/commit/b52b5877a580f64b5b6c7a5e6b54b5f1d4987bfb) covering all these fields, but I thought you might find our comparison scripts useful for other tools and such.

### Describe a possible solution to your feature or suggestion in detail

The following are columns that just have different names in AzerothCore and TrinityCore. In my opinion, for some of these your names are better so I wouldn't really suggest changing those. Maybe we can convince trinity to change theirs in those instances.

### broadcast_text
`MaleText` -> `Text`
`FemaleText` -> `Text1`
`SoundEntriesId` -> `SoundEntriesID`

### broadcast_text_locale
`MaleText` -> `Text`
`FemaleText` -> `Text1`

### game_event_battleground_holiday
`eventEntry` -> `EventEntry`

### npc_text
(for x 0-7 and y 0-5):
`em{x}_{y}` -> `Emote{x}_{y}`

### player_factionchange_items
`alliance_comment` -> `commentA`
`horde_comment` -> `commentH`

### spell_enchant_proc_data
`entry` -> `EnchantID`
`customChance` -> `Chance`
`PPMChance` -> `ProcsPerMinute`
`procEx` -> `AttributesMask`

### spell_proc
`spellId` -> `SpellId`
 `schoolMask`  -> `SchoolMask`
`spellFamilyName` -> `SpellFamilyName`
`spellFamilyMask0` -> `SpellFamilyMask0`
`spellFamilyMask1` -> `SpellFamilyMask1`
`spellFamilyMask2` -> `SpellFamilyMask2`
`spellTypeMask` -> `SpellTypeMask`
`spellPhaseMask` -> `SpellPhaseMask`
`hitMask` -> `HitMask`
`attributesMask` -> `AttributesMask`
`ratePerMinute` -> `RatePerMinute`
`chance` -> `Chance`
`cooldown` -> `Cooldown`
`charges` -> `Charges`

### achievement_reward_locale
`Text` -> `Body`

### AC rev. hash/commit
https://github.com/tswow/azerothcore-wotlk/commit/ee442e2f571eba480e92418c9fce32e723ca0973

### Additional context

Here is the [script i used to find these differences](https://gist.github.com/ihm-tswow/8700eda73ca0d33fd88b42874163c91c). It can detect missing tables, missing columns and type discrepancies between columns. Note that it does not detect different column orders, as we didn't need it for our application. You can modify the script to enable this if you need it.

To run the comparator, take the following steps:

1. Place the `dbccomparator.js` script in an empty directory
2. Edit the `DB_1` and `DB_2` variables to the names of your trinitycore/azerothcore database names.
3. In the directory where you placed the script, run `npm i mysql` to install the adapter
4. In the same directory, run `node dbcomparator` to compare the databases.
5. The output will show you missing tables, columns and type discrepancies between existing columns.

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://app.bountysource.com/issues/106210514-feature-align-db-column-names-to-trinitycore?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://app.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by heyitsbench**
## Changes Proposed:
- Adjusts table/column names in auth, characteres, and world DBs to [recent SQL standards](https://www.azerothcore.org/wiki/sql-standards#naming-of-tables-and-columns), as well as subjectively improving various column names.

## Issues Addressed:

- Closes #11005

## SOURCE:

N/A

## Tests Performed:

- Tested each query individually to make sure no errors occurred on my local DBs.

## How to Test the Changes:

1. Run the queries.
2. Observe that the auth/worldservers load properly and nothing changes in-game.

## Known Issues and TODO List:

- This is obviously not done. I'm opening this now so that any feedback on table/column name changes have time to be seen by contributors/reviewers/maintainers alike, and any changes that should be made can be made. The SQL queries don't have any work left to do on my end unless changes are requested, all that's left is adjusting the names in queries within the core.
- [X] Auth DB rename query.
- [X] Characters DB rename query.
- [X] World DB rename query.
- [X] Auth DB query core adjustments.
- [x] Characters DB query core adjustments.
- [ ] World DB query core adjustments.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by heyitsbench** [comment]
Also of note, this does include the changes proposed in [this PR](https://github.com/azerothcore/azerothcore-wotlk/pull/13058).

**Comment by Kitzunu** [comment]
Exclude broadcast_text and make a seperate PR for that

**Comment by heyitsbench** [comment]
> Exclude broadcast_text and make a seperate PR for that

[Done](https://github.com/azerothcore/azerothcore-wotlk/pull/14509).

**Comment by heyitsbench** [comment]
May be rather obvious, but this will likely bork most if not all parts of [Keira](https://github.com/azerothcore/Keira3). I can update modules and the wiki in time, but I am very unfamiliar with Keira, so it probably won't be me who updates that.

**Comment by Helias** [comment]
I do not dislike this renaming, but it will break Keira3, AoWoW and many other applications 😢 
Moreover, we will differ from TC a lot after this, which is a little benefit

I do not remember if the fields name on TC 8-10.x (or our current one) are sniffed from the retail, so they probably match with the official/retail fields name

Are we sure to do this?

**Comment by FrancescoBorzi** [comment]
I agree with @Helias , sorry but I don't think the amount of work here is worth the benefit. There are many other things that are more important for the project.

@heyitsbench if you still want to do this please prepare a fix also for all the tools such as Keira3, aowow, etc... but honestly I would not do it.

File: data/sql/updates/pending_db_auth/redo-auth.sql [hunk 1]
```diff
@@ -0,0 +1,117 @@
+ALTER TABLE `account`
```
Comment (FrancescoBorzi) [inline data/sql/updates/pending_db_auth/redo-auth.sql:1]: PR blocked until there is an equivalent PR for all the main tools depending on all the tables changed here. And it's a lot of work.

**Review by FrancescoBorzi — CHANGES_REQUESTED** [review]

**Comment by Nyeriah** [comment]
I thought this PR aligned the names with TC. If that’s not the case then I don’t think that’s something we’re looking for. Otherwise, could try breaking the PR by parts so there aren’t too many renames to address

**Comment by Helias** [comment]
this could break also several modules unfortunately

**Comment by heyitsbench** [comment]
I'm closing this for now. After further discussion, I will open a much smaller PR to include the changes in [this PR](https://github.com/azerothcore/azerothcore-wotlk/pull/13058), as well as align a few column names to TC, but I will not continue working on aligning the DB to a general consistent standard, due to preference of parity with TC.

**Pull request closed without merging**